### ISSUE TITLE
Report complex conditions on if statements using negative logic

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#543]
In case we have a combined complex `if` statement that uses negative logic inside its conditions as well, we should report a warning and propose to invert the `if`.

For example, following code should trigger a violation
```c#
if (!(parameter is string) || !char.TryParse((string)parameter, out value))
{
    DoSomething();
}
```

That code would be much easier to read and understand when inverted:
```c#
if (parameter is string && char.TryParse((string)parameter, out value))
{
    // TODO: Add comment here
}
else
{
    DoSomething();
}
```

**Comment by RalfKoban** [comment]
Same would be for:

```c# 
if (!(parameter is string s))
{
    DoSomething();
}
```
or
```c#
if (!(some.Name == "something" || IsSomething(some)))
```

**Comment by RalfKoban** [comment]
Same would be for:
```c#
if (!data.Values.Any(_ => Equals((T)_.Value, value)))
```

**Comment by RalfKoban** [comment]
Same would be for:
```c#
if (!_verifier.CanDoSomething(GetSomeData(obj.SomeInfo), items.Select(_ => _.Whatever.Data).ToList()))
```
